invenio_search/ext.py

```diff
@@ -69,23 +75,29 @@ def current_suffix(self):
             self._current_suffix = timestamp_suffix()
         return self._current_suffix
 
-    @cached_property
-    def templates(self):
-        """Generate a dictionary with template names and file paths."""
+    def _get_property(self, property_list):
```

Comment:
```suggestion
    def _collect_templates(self, entrypoint_group):
```

nit: two points for readablity:

- I'm somewhat confused by the naming. Since this method always gets information from entrypoints, I would use equivalent terminology.
- Another idea would be to merge `_get_property` and `load_entry_point_group_templates`, since the latter is not called from anywhere else. If we want to split them for avoiding the deep nesting, fine as well, but I would move `load_entry_point_group_templates` below `_get_property` (to keep related code together)